### PR TITLE
workflow to deploy to test environment

### PR DESCRIPTION
**Pull request by christopher-maboh**
## 🎫 Ticket

https://jira.cms.gov/browse/BCDA-PLT-187

## 🛠 Changes

(What was added, updated, or removed in this PR.)
Workflow to deploy to test environment added
## ℹ️ Context for reviewers

(Background context, more in-depth details of the implementation, and anything else you'd like to call out or ask reviewers.)
Create a GitHub Actions workflow in the bcda-app repo to deploy application changes to the test environment on merge to the main branch.
## ✅ Acceptance Validation

(How were the changes verified? Did you fully test the acceptance criteria in the ticket? Provide reproducible testing instructions and screenshots if applicable.)

## 🔒 Security Implications

- [ ] This PR adds a new software dependency or dependencies.
- [ ] This PR modifies or invalidates one or more of our security controls.
- [ ] This PR stores or transmits data that was not stored or transmitted before.
- [ ] This PR requires additional review of its security implications for other reasons.

If any security implications apply, add Jason Ashbaugh (GitHub username: StewGoin) as a reviewer and do not merge this PR without his approval.

### CODE REVIEW

File: .github/workflows/deploy-to-test.yml [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v4
+
+      - name: Decrypt secrets
+        run: |
+          echo $VAULT_PW > .vault_password
+          bash ops/secrets --decrypt
+          mv -fv shared_files/encrypted/* shared_files/decrypted/
```
Comment (gsf) [inline .github/workflows/deploy-to-test.yml:28]: We might be able to switch out this secrets step to use AWS Parameter Store instead.

File: .github/workflows/deploy-to-test.yml [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+      - name: configure aws credentials
+        uses: aws-actions/configure-aws-credentials@v3
+        with:
+          role-to-assume: arn:aws:iam::${{ secrets.AWS_ACCOUNT_NUMBER }}:role/github-to-aws-oidc
+          aws-region: us-east-1
+
+      - name: Deploy to Test Environment
+        run: |
+          # Stop the existing containers if any
+          docker-compose down
```
Comment (gsf) [inline .github/workflows/deploy-to-test.yml:42]: Should be in cleanup if needed (maybe not given short-lived self-hosted, or definitely not if on github hosted infra)

File: .github/workflows/deploy-to-test.yml [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+
+          # Start the application in the test environment
+          docker-compose up -d
```
Comment (gsf) [inline .github/workflows/deploy-to-test.yml:45]: TODO: deploy to test environment instead of running locally

**Comment by carlpartridge** [comment]
@gsf @christopher-maboh Is this still relevant?  I am unable to see the PLT ticket.

**Pull request closed without merging**